flickit-assessment-kit/src/main/java/org/flickit/assessment/kit/adapter/out/persistence/kitversion/KitVersionMapper.java

```diff
@@ -35,4 +36,8 @@ public static KitVersion mapToDomainModel(KitVersionJpaEntity entity) {
             entity.getCreatedBy(),
             entity.getLastModifiedBy());
     }
+
+    public static long mapKitVersionToVersionStatus(KitVersionStatus status, long id){
+        return status == KitVersionStatus.ARCHIVE ? -id : status.getId();
```

Comment:
**Suggestion:** Modify the `mapKitVersionToVersionStatus` method to handle potential edge cases where `status` or `id` could lead to unexpected negative values, ensuring the logic is clear and covers all relevant cases. [general, importance: 7]
```suggestion
        public static long mapKitVersionToVersionStatus(KitVersionStatus status, long id){
            if (status == null || id <= 0) throw new IllegalArgumentException("Invalid status or id");
            return status == KitVersionStatus.ARCHIVE ? -id : status.getId();
        }
```